src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -428,6 +432,16 @@ class MySimulation : public SimulationHelper<simDim>
         ForEach<VectorSpeciesWithIonizer, particles::CallIonization<bmpl::_1>, MakeIdentifier<bmpl::_1> > particleIonization;
         particleIonization(forward(particleStorage), cellDescription, currentStep);
 
+        /* SynchrotronPhotons */
+        typedef typename PMacc::particles::traits::FilterByFlag<VectorAllSpecies,
+                                                                synchrotronPhotons<> >::type AllSynchrotronPhotonsSpecies;
+
+        ForEach<AllSynchrotronPhotonsSpecies,
+                particles::CallSynchrotronPhotons<bmpl::_1>,
+                MakeIdentifier<bmpl::_1> > electronSynchrotronPhotons;
+        electronSynchrotronPhotons(forward(particleStorage), cellDescription, currentStep, this->synchrotronFunctions);
```

Comment:
If you make `this->synchrotronFunctions` only available if `electronSynchrotronPhotons` is not the empty list, is it still required to resolve `this->synchrotronFunctions`? If so, you need to deal with this here via a `#ifdefine` or similar.